### PR TITLE
Added Requirements and Optional Dependencies to the readme.

### PR DESCRIPTION
**Pull request by Nessworthy**
Addresses #91.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -5,13 +5,25 @@ Chat bot for StackOverflow. Uses [amphp](https://github.com/amphp) libraries for
...
 ## Installation
 
 1. Clone the project.
 1. Copy `config/config.sample.yml` to `config/config.yml`.
 1. Replace all configuration variables with your values.
 1. Run the bot using `php ./cli/run.php`.
 
+## Optional Dependencies
+
+1. For true non-blocking execution:
```
Comment (staabm) [inline README.md]: You either need uv or ev not both

File: README.md [hunk 2]
```diff
@@ -5,13 +5,25 @@ Chat bot for StackOverflow. Uses [amphp](https://github.com/amphp) libraries for
 
 [![Build Status](https://travis-ci.org/Room-11/Jeeves.svg?branch=master)](https://travis-ci.org/Room-11/Jeeves)
 
+## Requirements
```
Comment (staabm) [inline README.md]: Shouldnt those just be declared in composer.json

**Comment by Nessworthy** [comment]
Cheers @staabm, I'll crack on with those changes tonight.

File: README.md (hunk 2 above)
Comment (DaveRandom) [inline README.md]: They are. It doesn't hurt to put it in the readme I suppose, but it does mean there are two places to update if deps are added/removed.

Comment (PeeHaa) [inline README.md]: I for one ~~~welcome our new README overlords~~~ always check the README first for requirements.

It means having to maintain it twice but I personally can live with that.

File: README.md [hunk 3]
```diff
@@ -5,13 +5,25 @@ Chat bot for StackOverflow. Uses [amphp](https://github.com/amphp) libraries for
...
+
+1. PHP 7.0 or greater.
+1. Ensure the `php_intl` extension is installed and enabled. [Read the PHP Documentation](http://php.net/manual/en/book.intl.php) for more information.
+
```
Comment (PeeHaa) [inline README.md]: IIRC @fapore is running CentOS 6

**Comment by PeeHaa** [comment]
BTW not exactly related to #91. But it would be nice to also have a mention to the license in the readme I guess?

**Comment by Nessworthy** [comment]
What license does Jeeves actually use?

File: README.md (hunk 2 above)
Comment (kelunik) [inline README.md]: :+1: for requirements in README

**Comment by PeeHaa** [comment]
MIT @Nessworthy

**Comment by PeeHaa** [comment]
Thanks @Nessworthy!

**Comment by Nessworthy** [comment]
👍

**Pull request merged**